src/Catalog/Helpers/Utils.cs

```diff
@@ -476,36 +476,47 @@ public static IEnumerable<PackageEntry> GetEntries(ZipArchive package)
             return result;
         }
 
-        public static NupkgMetadata GetNupkgMetadata(Stream stream)
+        public static NupkgMetadata GetNupkgMetadata(Stream stream, string packageHash)
         {
-            var nupkgMetadata = new NupkgMetadata
+            if (stream == null)
             {
-                PackageSize = stream.Length,
-                PackageHash = GenerateHash(stream)
-            };
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            var packageSize = stream.Length;
+
+            packageHash = packageHash ?? GenerateHash(stream);
 
             stream.Seek(0, SeekOrigin.Begin);
 
-            using (ZipArchive package = new ZipArchive(stream, ZipArchiveMode.Read, true))
+            using (var package = new ZipArchive(stream, ZipArchiveMode.Read, leaveOpen: true))
             {
-                nupkgMetadata.Nuspec = GetNuspec(package);
+                var nuspec = GetNuspec(package);
```

Comment:
we can do the same for Catalog2Dnx :)

Reply:
Do what exactly?  Sorry, I'm not following.